easybuild/tools/modules.py

```diff
@@ -738,18 +738,30 @@ def modpath_extensions_for(self, mod_names):
         # regex for $MODULEPATH extensions;
         # via 'module use ...' or 'prepend-path MODULEPATH' in Tcl modules,
         # or 'prepend_path("MODULEPATH", "...") in Lua modules
+        modpath_ext_use_regex = '^\s*module\s+use\s+"?([^"\s]+)"?'  # 'module use' in Tcl module files
         modpath_ext_regex = r'|'.join([
-            r'^\s*module\s+use\s+"?([^"\s]+)"?',  # 'module use' in Tcl module files
+            modpath_ext_use_regex,
             r'^\s*prepend-path\s+MODULEPATH\s+"?([^"\s]+)"?',  # prepend to $MODULEPATH in Tcl modules
             r'^\s*prepend_path\(\"MODULEPATH\",\s*\"(\S+)\"',  # prepend to $MODULEPATH in Lua modules
         ])
+        modpath_ext_use_regex = re.compile(modpath_ext_use_regex, re.M)
         modpath_ext_regex = re.compile(modpath_ext_regex, re.M)
 
         modpath_exts = {}
         for mod_name in mod_names:
+            # we must include parsed MODULEPATH extensions where the module tool expands
+            # environment variables, concatenates strings, etc.
+            # The show output is actually sufficient except for modulecmd.tcl < 1.661
```

Comment:
I'm not sure if this is correct, I think we still want to include `$MODULEPATH` extensions that are guarded, cfr. my other comment: if the directory is created at a later point, all of a sudden we'll have a module A again that loads a module B that extend the `$MODULEPATH` with the location where module A resides, which doesn't make sense...